chrome/content/zotero/progressQueueDialog.js

```diff
@@ -217,3 +220,20 @@ var Zotero_RecognizePDF_Dialog = new function () {
 		}
 	}
 };
+
+/**
+ * Create a dialog for each progressQueue
+ */
+var Zotero_ProgressQueue_Dialogs = new function () {
```

Comment:
Should this be Zotero.ProgressQueueDialogs within xpcom instead? The dialog isn't tied to a particular window, and if we allowed multiple Zotero windows you'd want to be able to access a dialog from the toolbar button in any window.

Reply:
One problem is that when xpcom is initializing the dialogs are not available at the time. Therefore we have to wait until UI part is initialized. Unless there is a workaround.

Reply:
Actually I think the dialog should be in xpcom/ too (similar to xpcom/progressWindow.js, the bottom-right save popups). We can just get rid of Zotero_ProgressQueue_Dialogs and move getDialog() to the Zotero.ProgressQueue object, and then lazily initialize a Zotero.ProgressQueueDialog when getDialog() is called.

Reply:
For opening a window within xpcom, whether or not there's a frontmost window:

```javascript
var win = Services.wm.getMostRecentWindow("navigator:browser");
if (win) {
	_progressWindow = win.openDialog("chrome://zotero/content/progressQueueDialog.xul",
		"", "chrome,close=yes,resizable=yes,dependent,dialog,centerscreen");
}
else {
	_progressWindow = Services.ww.openWindow(null, "chrome://zotero/content/progressQueueDialog.xul",
		"", "chrome,close=yes,resizable=yes,dependent,dialog,centerscreen", null);
}
```

(Can do it later, but we should put this in a zotero.js function at some point, next to getActiveZoteroPane.)